Sources/Swiftly/Init.swift

```diff
@@ -0,0 +1,206 @@
+            """)
+
+            if SwiftlyCore.readLine(prompt: "> ") == "0" {
+                throw Error(message: "Swiftly installation has been cancelled")
+            }
+        }
+
+        let shell = if let s = ProcessInfo.processInfo.environment["SHELL"] {
+            s
+        } else {
+            try await Swiftly.currentPlatform.getShell()
+        }
+
+        let envFile: URL
+        let sourceLine: String
+        if shell.hasSuffix("fish") {
+            envFile = Swiftly.currentPlatform.swiftlyHomeDir.appendingPathComponent("env.fish", isDirectory: false)
+            sourceLine = """
+
+            # Added by swiftly
+            source "\(envFile.path)"
+            """
+        } else {
+            envFile = Swiftly.currentPlatform.swiftlyHomeDir.appendingPathComponent("env.sh", isDirectory: false)
+            sourceLine = """
+
+            # Added by swiftly
+            . "\(envFile.path)"
+            """
+        }
+
+        if overwrite {
+            try? FileManager.default.removeItem(at: Swiftly.currentPlatform.swiftlyToolchainsDir)
+            try? FileManager.default.removeItem(at: Swiftly.currentPlatform.swiftlyHomeDir)
+        }
+
+        // Go ahead and create the directories as needed
+        for requiredDir in Swiftly.requiredDirectories {
+            if !requiredDir.fileExists() {
+                do {
+                    try FileManager.default.createDirectory(at: requiredDir, withIntermediateDirectories: true)
+                } catch {
+                    throw Error(message: "Failed to create required directory \"\(requiredDir.path)\": \(error)")
+                }
+            }
+        }
+
+        // Force the configuration to be present. Generate it if it doesn't already exist or overwrite is set
+        if overwrite || config == nil {
+            let pd = try await Swiftly.currentPlatform.detectPlatform(disableConfirmation: assumeYes, platform: platform)
+            var config = Config(inUse: nil, installedToolchains: [], platform: pd)
+            // Stamp the current version of swiftly on this config
+            config.version = SwiftlyCore.version
+            try config.save()
+        }
+
+        let swiftlyBin = Swiftly.currentPlatform.swiftlyBinDir.appendingPathComponent("swiftly", isDirectory: false)
+
+        let cmd = CommandLine.arguments[0]
+        let systemManaged = try Swiftly.currentPlatform.isSystemManagedBinary(cmd)
+
+        // Don't move the binary if this is being invoked inside an xctest or is a system managed binary
+        if !cmd.hasSuffix("xctest") && !systemManaged {
+            SwiftlyCore.print("Moving swiftly into the installation directory...")
+            if swiftlyBin.fileExists() && !overwrite {
+                throw Error(message: "Swiftly binary already exists. You can try again with overwrite to replace it.")
+            }
+            do {
+                try FileManager.default.moveItem(at: URL(fileURLWithPath: cmd), to: swiftlyBin)
```

Comment:
`moveItem` fails if the item exists already. Also should we check that cmd and swiftlyBin aren't exactly the same

Reply:
Thanks, I had assumed that the move and copy would overwrite. I've made an explicit remove in the case of overwrite and there is an existing binary.